gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePlugin.java

```diff
@@ -215,9 +223,27 @@ private void setupErrorProneOptions(
                 public Iterable<String> asArguments() {
                     List<String> suppressionsToRemove = checksToRemoveSuppressionsFor(project);
 
+                    Set<String> checksToPatch = new HashSet<>();
+                    checksToPatch.add("RemoveRolloutSuppressions");
+
+                    // If we're also applying suggested patches, we want to make sure that these are a subset of
+                    //   the checks we're removing suppressions for.
+                    // Otherwise, we might apply fixes for a check that we're not removing the suppression for, if
+                    //   suppressed using a for-rollout suppression (because we're not applying the for-rollout hack)
+                    if (isApplyingSuggestedPatches(project)) {
+                        Set<String> suppressionsToRemoveSet = new HashSet<>(suppressionsToRemove);
+                        List<String> extraChecksToPatch =
+                                checksToApplySuggestedPatchesFor(extension, javaCompile, errorProneOptions);
+                        if (extraChecksToPatch.stream().anyMatch(check -> !suppressionsToRemoveSet.contains(check))) {
+                            throw new IllegalStateException(
+                                    "Checks to patch must be a subset of the checks to remove suppressions for");
+                        }
```

Comment:
I'd actually collect the checks that aren't in the remove set to a list and check it's empty. Then we can paste all the ones that are not there in the error message (I don't really like error messages that don't show this kind of info).

Reply:
Yeah, fair enough!